### PR TITLE
Support secondary only connections

### PR DESCRIPTION
**Pull request by manover**
Allow option of connections to secondary only members of replica set

### CODE REVIEW

**Comment by manover** [comment]
@jehiah Ready for review

File: asyncmongo/asyncjobs.py [hunk 1]
```diff
@@ -125,17 +128,25 @@ def process(self, response=None, error=None):
...
             logging.debug("Working through the rest of the host list: %r", fresh)
 
             while fresh:
-                if self._primary and self._primary not in self._tried_hosts:
+                if self._primary and self._primary not in self._blacklisted:
                     # Try primary first
                     h = self._primary
                 else:
-                    h = fresh.pop()
+                    h = random.sample(fresh, len(fresh))[0]
```
Comment (jehiah) [inline asyncmongo/asyncjobs.py]: is this any different from `random.choice(fresh)` ?

File: asyncmongo/connection.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ class Connection(object):
       - `autoreconnect` (optional): auto reconnect on interface errors
       - `rs`: replica set name (required when replica sets are used)
       - `seed`: seed list to connect to a replica set (required when replica sets are used)
+      - `secondary`: (optional, only useful for replica set connections) if true, connect to a secondary member
```
Comment (jehiah) [inline asyncmongo/connection.py]: I think the "only" part is important here (present in other variables).  what about `secondary_only` or `secondary_instances_only`. bordering on verbose, but i think clarity trumps.

**Comment by jehiah** [comment]
@manover nice work on this change. aside from some small renames, :thumbsup:

File: asyncmongo/asyncjobs.py (hunk 1 above)
Comment (manover) [inline asyncmongo/asyncjobs.py]: actually ... you are right, I was thinking about consuming the sequence all the time, but have come to picking a random member, so choice() will be a much cleaner way to do this, thanks!

File: asyncmongo/connection.py (hunk 2 above)
Comment (manover) [inline asyncmongo/connection.py]: agreed

**Pull request merged**